tests/test_formats.py

```diff
@@ -1060,6 +1062,511 @@ def test_format_number_with_sep_dec_marks():
+                "10:10:30 PM",
+                "11:15:00 PM",
+            ],
+        ),
+        (
+            "h_m_p",
+            ["12:00 AM", "8:30 AM", "10:59 AM", "12:00 PM", "1:23 PM", "10:10 PM", "11:15 PM"],
+        ),
+        ("h_p", ["12 AM", "8 AM", "10 AM", "12 PM", "1 PM", "10 PM", "11 PM"]),
+    ],
+)
+def test_fmt_time(time_style: TimeStyle, x_out: str):
+    df = pd.DataFrame(
+        {
+            "x": [
+                "00:00:00",
+                "08:30:00",
+                "10:59:59",
+                "12:00:00",
+                "13:23:59",
+                "22:10:30",
+                "23:15",
+            ]
+        }
+    )
+
+    gt = GT(df).fmt_time(columns="x", time_style=time_style)
+    x = _get_column_of_values(gt, column_name="x", context="html")
+    assert x == x_out
+
+
+# ------------------------------------------------------------------------------
+# Test `fmt_datetime()`
+# ------------------------------------------------------------------------------
+
+
+@pytest.mark.parametrize(
+    "date_style,time_style,sep,x_out",
+    [
+        (
+            "iso",
+            "iso",
+            " ",
+            [
+                "2023-01-05 00:00:00",
+                "2022-06-15 08:30:00",
+                "2020-11-12 10:59:59",
+                "2015-04-25 12:00:00",
+                "2016-12-04 13:23:59",
+                "2018-07-28 22:10:30",
+                "2013-05-15 23:15:00",
+            ],
+        ),
+        (
+            "wday_month_day_year",
+            "h_m_s_p",
+            " at ",
+            [
+                "Thursday, January 5, 2023 at 12:00:00 AM",
+                "Wednesday, June 15, 2022 at 8:30:00 AM",
+                "Thursday, November 12, 2020 at 10:59:59 AM",
+                "Saturday, April 25, 2015 at 12:00:00 PM",
+                "Sunday, December 4, 2016 at 1:23:59 PM",
+                "Saturday, July 28, 2018 at 10:10:30 PM",
+                "Wednesday, May 15, 2013 at 11:15:00 PM",
+            ],
+        ),
+    ],
+)
+def test_fmt_datetime(date_style: DateStyle, time_style: TimeStyle, sep: str, x_out: str):
+    df = pd.DataFrame(
+        {
+            "x": [
+                "2023-01-05 00:00:00",
+                "2022-06-15 08:30:00",
+                "2020-11-12 10:59:59",
+                "2015-04-25 12:00:00",
+                "2016-12-04 13:23:59",
+                "2018-07-28 22:10:30",
+                "2013-05-15 23:15",
+            ]
+        }
+    )
+
+    gt = GT(df).fmt_datetime(columns="x", date_style=date_style, time_style=time_style, sep=sep)
+    x = _get_column_of_values(gt, column_name="x", context="html")
+    assert x == x_out
+
+
+# ------------------------------------------------------------------------------
+# Test `fmt_bytes()`
+# ------------------------------------------------------------------------------
+
+
+@pytest.mark.parametrize(
+    "standard,decimals,use_seps,force_sign,incl_space,x_out",
```

Comment:
See feedback on avoiding using many parameters, and resetting things to their default arguments.

See also feedback on using 1 long input across all cases. I can't tell which values in the output tell us something new. For example all the outputs contain "-342 B", so it's not clear what I should learn there across the 5 parameter set cases.

(Sometimes tests do need to validate that some result is unchanged, but it seems likely the testing here is maybe doing that repeatedly for the same underlying logic)